src/main/java/com/eleks/academy/whoami/core/state/SuggestingCharacters.java

```diff
@@ -40,7 +41,12 @@ public boolean finished() {
 	public void setCharacters(String player, CharacterSuggestion character) {
 		SynchronousPlayer findPlayer = findPlayer(player)
 				.orElseThrow(() -> new HttpClientErrorException(HttpStatus.NOT_FOUND, "Player not found"));
-		findPlayer.suggestCharacter(character);
+		if (findPlayer.getNickName() != null) {
```

Comment:
please add tests that these if-else blocks are covered

Reply:
done